packages/drivers/routerlicious-driver/src/test/r11sSocketTests.spec.ts

```diff
@@ -194,4 +195,111 @@ describe("R11s Socket Tests", () => {
 			"Error internal code should be R11sServiceClusterDrainingErrorCode",
 		);
 	});
+
+	it("Socket error with Data Corruption error", async () => {
+		const errorToThrow = {
+			code: 500,
+			message: "Data corruption detected",
+			retryAfterMs: 0,
+			internalErrorCode: "DataCorruption",
+			errorType: FluidErrorTypes.dataCorruptionError,
+			canRetry: false,
+		};
+		const errorEventName = "connect_document_success";
+		socket = new ClientSocketMock({
+			connect_document: { eventToEmit: errorEventName },
+		});
+
+		const connection = await mockSocket(socket as unknown as Socket, async () =>
+			documentService.connectToDeltaStream(client),
+		);
+		let error: IAnyDriverError | undefined;
+		connection.on("disconnect", (reason: IAnyDriverError) => {
+			error = reason;
+		});
+
+		// Send Data Corruption error
+		socket.sendErrorEvent(errorToThrow);
+
+		assert(
+			error?.errorType === FluidErrorTypes.dataCorruptionError,
+			"Error type should be dataCorruptionError",
+		);
+		assert(error.scenarioName === "error", "Error scenario name should be error");
+		assert(
+			(error as any).internalErrorCode === "DataCorruption",
+			"Error internal code should be DataCorruption",
+		);
+	});
+
+	it("Socket error with Data Processing error", async () => {
+		const errorToThrow = {
+			code: 500,
+			message: "Data processing error",
+			retryAfterMs: 0,
+			internalErrorCode: "DataProcessing",
+			errorType: FluidErrorTypes.dataProcessingError,
+			canRetry: false,
+		};
+		const errorEventName = "connect_document_success";
+		socket = new ClientSocketMock({
+			connect_document: { eventToEmit: errorEventName },
+		});
+
+		const connection = await mockSocket(socket as unknown as Socket, async () =>
+			documentService.connectToDeltaStream(client),
+		);
+		let error: IAnyDriverError | undefined;
+		connection.on("disconnect", (reason: IAnyDriverError) => {
+			error = reason;
+		});
+
+		// Send Data Processing error
+		socket.sendErrorEvent(errorToThrow);
```

Comment:
Again, based on my understanding of the ClientSocketMock, the ClientSocketMock is acting as the server here. So, instead of `socket.sendErrorEvent("data corruption detected by client")` we should instead be doing something like

```
// PSEUDO CODE
clientError = "data corruption detected by client"
deltaStreamConnection.disconnect(reason = clientError)
assert(socket.received(eventName = "disconnect_document", reason = clientError);
```